payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -318,13 +321,30 @@ internal class DefaultCardNumberController(
         previousFocusDirection: FocusDirection
     ) {
         val reporter = LocalCardNumberCompletedEventReporter.current
+        val disallowedBrandReporter = LocalCardBrandDisallowedReporter.current
+
+        // Remember the last state indicating whether it was a disallowed card brand error
+        val lastLoggedCardBrand = remember { mutableStateOf<CardBrand?>(null) }
```

Comment:
Should we do this inside the reporter, rather than the UI?

Reply:
If not, you probably want to change the syntax here a bit. Looks like you want a `var lastLoggedCardBrand by`

Reply:
You'll also probably want a `rememberSaveable`, which will ensure that rotation (or any configuration change) doesn't cause another execution.

Reply:
Ah didn't know that, thanks! I've updated this to be a `by rememberSaveable`